### PR TITLE
Remove overloaded upstream files and tools

### PR DESCRIPTION
**Pull request by krzysz00**
This PR arises from discussion in chat yesterday about how we shouldn't be overriding upstream's mlir-opt and such.

### CODE REVIEW

**Comment by whchung** [comment]
Looks fine to me.

**Comment by krzysz00** [comment]
@jungpark-mlir - this is touching MIGraphX stuff, you might want a look.

File: mlir/include/mlir/InitMIOpenPasses.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+// This file defines a helper to trigger the registration of all our custom
+// dialects and passes to the system.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_INITMIOPENPASSES_H_
+#define MLIR_INITMIOPENPASSES_H_
+
+#include "mlir/Conversion/MIOpenPasses.h"
+#include "mlir/Dialect/MIGraphX/Passes.h"
```
Comment (jungpark-mlir) [inline mlir/include/mlir/InitMIOpenPasses.h]: Sorry, this is dead code. migraphx dialect doesn't have any pass.
include/...../Dialect/MIGraphX/pass.h, pass.td should be deleted and related part in the cmakefile also needs to be removed.
Of course, no pass to register.

File: mlir/include/mlir/InitMIOpenPasses.h [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+// global registry.
+// If you're building a compiler, you likely don't need this: you would build a
+// pipeline programmatically without the need to register with the global
+// registry, since it would already be calling the creation routine of the
+// individual passes.
+// The global registry is interesting to interact with the command-line tools.
+inline void registerMIOpenPasses() {
+  registerMIOpenConversionPasses();
+  miopen::registerPasses();
+  migraphx::registerPasses();
```
Comment (jungpark-mlir) [inline mlir/include/mlir/InitMIOpenPasses.h]: not needed

File: mlir/lib/Dialect/MIGraphX/MiGraphXPasses.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
+//===- MIGraphXPasses.cpp - MIGraphX MLIR Passes
```
Comment (jungpark-mlir) [inline mlir/lib/Dialect/MIGraphX/MiGraphXPasses.cpp]: file needs to be deleted

**Review by jungpark-mlir — CHANGES_REQUESTED** [review]
I should have cleaned this up before but appreciate if you can delete it here.

File: mlir/include/mlir/InitMIOpenPasses.h (hunk 1 above)
Comment (jungpark-mlir) [reply mlir/include/mlir/InitMIOpenPasses.h]: Just found mlir/lib/Dialect/MIGraphX/PassDetail.h also needs to be deleted
and its cmakefile needs to remove the file and dependency to the MLIRMIGraphXPassIncGen

Comment (krzysz00) [reply mlir/include/mlir/InitMIOpenPasses.h]: @jungpark-mlir - I've gone and deleted all the stuff

**Review by jungpark-mlir — APPROVED** [review]

**Pull request merged**